climada_petals/engine/warn.py

```diff
@@ -0,0 +1,367 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define the Warn class.
+"""
+import logging
+import copy
+from dataclasses import dataclass, field
+
+import numpy as np
+import skimage
+from typing import List, Dict
+
+from climada.util.plot import geo_scatter_categorical
+
+LOGGER = logging.getLogger(__name__)
+
+
+def dilation(bin_map, size):
+    """Dilate binary input map. Enlarges and connects regions of interests. Larger sizes - more area of interest.
+
+    Parameters
+    ----------
+    bin_map : np.ndarray
+        Rectangle 2d map of values which are used to generate the warning.
+    size : int
+        Size of kernel.
+
+    Returns
+    ----------
+    np.ndarray
+        Generated binary map with enlarged regions of interest.
+    """
+    return skimage.morphology.dilation(bin_map, skimage.morphology.disk(size))
+
+
+def erosion(bin_map, size):
+    """Erode binary input map. Reduces region of interest and heterogeneity in map. Larger sizes - more reduction.
+
+    Parameters
+    ----------
+    bin_map : np.ndarray
+        Rectangle 2d map of values which are used to generate the warning.
+    size : int
+        Size of kernel.
+
+    Returns
+    ----------
+    np.ndarray
+        Generated binary map with reduced regions of interest.
+    """
+    return skimage.morphology.erosion(bin_map, skimage.morphology.disk(size))
+
+
+def median_filtering(bin_map, size):
+    """Smooth out binary map. Larger sizes - smoother regions and more reduction of heterogeneity.
+
+    Parameters
+    ----------
+    bin_map : np.ndarray
+        Rectangle 2d map of values which are used to generate the warning.
+    size : int
+        Size of kernel.
+
+    Returns
+    ----------
+    np.ndarray
+        Generated binary map with smoothed regions of interest.
+    """
+    return skimage.filters.median(bin_map, np.ones((size, size)))
+
+
+from enum import Enum
+
+class Operation(Enum):
+    dilation = dilation
+    erosion = erosion
+    median_filtering = median_filtering
+    
+    def __call__(*args, **kwargs):  # this is not strictly necessary but with this one can do `Operation['dilation'](map, size)`
```

Comment:
Calling like this does not work, or I did something wrong?! 'Operation.dilation(map, size)' does work but we need it to be adjustable. Did I misunderstand something?